client/state/themes/actions.js

```diff
@@ -94,87 +93,49 @@ export function receiveTheme( theme, siteId ) {
 }
 
 /**
- * Returns an action object to be used in signalling that theme objects have
- * been received.
- *
- * @param  {Array}  themes Themes received
- * @param  {Number} siteId ID of site for which themes have been received
- * @return {Object}        Action object
- */
-export function receiveThemes( themes, siteId ) {
-	return ( dispatch, getState ) => {
-		const filterWpcom = shouldFilterWpcomThemes( getState(), siteId );
-		const { filteredThemes } = filterThemes( themes, siteId, filterWpcom );
-
-		dispatch( {
-			type: THEMES_RECEIVE,
-			themes: filteredThemes,
-			siteId
-		} );
-	};
-}
-
-/**
  * Returns an action object to be used in signalling that theme objects from
  * a query have been received.
  *
  * @param {Array}  themes Themes received
  * @param {number} siteId ID of site for which themes have been received
- * @param {Object} query Theme query used in the API request
- * @param {number} foundCount Number of themes returned by the query
+ * @param {?Object} query Theme query used in the API request
+ * @param {?number} foundCount Number of themes returned by the query
  * @return {Object} Action object
  */
-export function receiveThemesQuery( themes, siteId, query, foundCount ) {
+export function receiveThemes( themes, siteId, query, foundCount ) {
 	return ( dispatch, getState ) => {
-		const filterWpcom = shouldFilterWpcomThemes( getState(), siteId );
-		const { filteredThemes, found } = filterThemes(
-			themes,
-			siteId,
-			filterWpcom,
-			query,
-			foundCount,
-		);
+		let filteredThemes = themes;
+		let found = foundCount;
+
+		if ( siteId !== 'wporg' && siteId !== 'wpcom' ) {
```

Comment:
Now that it's inlined, maybe we should turn this check into an 

```js
if( isJetpackSite( getState(), siteId )
```

? That'd be a bit more semantic -- theoretically, `siteId` could be a numeric WP.com `siteId`, in which case we wouldn't want to filter. (Right now, we're relying on consumers to not pass a WP.com `siteId` here, but better safe than sorry, I guess...)